app/src/main/java/com/example/android/tvleanback/ui/PlaybackOverlayFragment.java

```diff
@@ -505,4 +481,40 @@ public void onItemClicked(Presenter.ViewHolder itemViewHolder, Object item,
         }
     }
 
+    private class MediaControllerCallback extends MediaController.Callback {
+        @Override
+        public void onPlaybackStateChanged(PlaybackState state) {
+            Log.d(TAG, "playback state changed: " + state.getState());
+            if (state.getState() == PlaybackState.STATE_PLAYING && mCurrentPlaybackState != PlaybackState.STATE_PLAYING) {
```

Comment:
THe problem is that the MediaSession.Playback State is not granular enough. So when a seekTo() is done on the VideoView, the new position has to be communicated to all MediaController.Callbacks which have to reflect the new position in the UI. So each time the position change we have to fire a playbackStateChange.

That's why it easily possible that onPlaybackStateChanged is called several times with PlaybackState.PLAYING because we have to update the new position only (state.getPosition()). That's why the mCurrentPlaybackState is needed to know when we have to start the progress automation.